src/snowflake/snowpark/modin/plugin/_internal/pivot_utils.py

```diff
@@ -870,7 +884,7 @@ def generate_single_pivot_labels(
             value_pandas_label_to_identifiers: Aggregation value pandas label to snowflake quoted identifier
             pandas_single_aggr_func: pandas aggregation function to apply to pandas aggregation label
     """
-    if isinstance(aggfunc, list):
+    if not is_dict_like(aggfunc) and is_list_like(aggfunc):
```

Comment:
why is the `not is_dict_like` qualifier needed?

Reply:
dictionaries are both `is_dict_like` and `is_list_like` :(